##
# @file This file is part of EDGE.
#
# @author Alexander Breuer (breuer AT mytum.de)
#
# @section LICENSE
# Copyright (c) 2019, Alexander Breuer
# All rights reserved.
#
# Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
#
# 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
#
# 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
#
# 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
#
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
#
# @section DESCRIPTION
# Patches the installation of EDGE's partition plugin into Gmsh.
##
--- a/Plugin/CMakeLists.txt
+++ b/Plugin/CMakeLists.txt
@@ -33,6 +33,7 @@ set(SRC
   CutMesh.cpp
   SpanningTree.cpp
   NewView.cpp
+  EdgePartition.cpp
   SimplePartition.cpp Crack.cpp
   MeshSubEntities.cpp
   MeshVolume.cpp
--- a/Plugin/PluginManager.cpp
+++ b/Plugin/PluginManager.cpp
@@ -24,6 +24,7 @@
 #include "MathEval.h"
 #include "ExtractElements.h"
 #include "SimplePartition.h"
+#include "EdgePartition.h"
 #include "Crack.h"
 #include "HarmonicToTime.h"
 #include "ModulusPhase.h"
@@ -261,6 +262,8 @@ void PluginManager::registerDefaultPlugins()
       "NewView", GMSH_RegisterNewViewPlugin()));
     allPlugins.insert(std::pair<std::string, GMSH_Plugin *>(
       "SimplePartition", GMSH_RegisterSimplePartitionPlugin()));
+    allPlugins.insert(std::pair<std::string, GMSH_Plugin *>(
+      "EdgePartition", GMSH_RegisterEdgePartitionPlugin()));
     allPlugins.insert(std::pair<std::string, GMSH_Plugin *>(
       "Crack", GMSH_RegisterCrackPlugin()));
     allPlugins.insert(std::pair<std::string, GMSH_Plugin *>(
